File: Java/2335. Minimum Amount of Time to Fill Cups.java
class Solution {
    public int fillCups(int[] amount) {
        int m, res = 0, c = amount[0], w = amount[1], h = amount[2];

        while(c > 0 || w > 0 || h > 0){
            m = Math.min(Math.min(c, w), h);

            if(m == c){
                w--;
                h--;
            }
            else if(m == w){
                c--;
                h--;
            }
            else if(m == h){
                c--;
                w--;
            }

            res++;
        }

        return res;
    }
}

File: Python/2335. Minimum Amount of Time to Fill Cups.py
class Solution:
    def fillCups(self, amount: List[int]) -> int:
        res = 0
        c, w, h = amount

        while c > 0 or w > 0 or h > 0:
            m = min(c, w, h)
            
            if m == c:
                w -= 1
                h -= 1
            elif m == w:
                c -= 1
                h -= 1
            elif m == h:
                c -= 1
                w -= 1
            
            res += 1

        return res
